hera_cal/abscal.py

```diff
@@ -981,510 +2636,319 @@ def TT_Phi_gain_arr(self):
+                           tol=1.0, phs_max_iter=100, phs_conv_crit=1e-6, clobber=True, add_to_history='', verbose=True):
+    '''Perform abscal on entire data files, picking relevant model_files from a list and doing partial data loading.
+    
+    Arguments:
+        data_file: string path to raw uvh5 visibility file
+        redcal_file: string path to calfits file that redundantly calibrates the data_file
+        model_files: list of string paths to externally calibrated data. Strings must be sortable 
+            to produce a chronological list in LST (wrapping over 2*pi is OK)
+        output_replace: tuple of two strings to find and replace in redcal_file to produce the output calfits file
+        nInt_to_load: number of integrations to load and calibrate simultaneously. Default -1 loads all integrations.
+        data_solar_horizon: Solar altitude threshold [degrees]. When the sun is too high in the data, flag the integration.
+        model_solar_horizon: Solar altitude threshold [degrees]. When the sun is too high in the model, flag the integration.
+        min_bl_cut: minimum baseline separation [meters] to keep in data when calibrating. None or 0 means no mininum,
+            which will include autocorrelations in the absolute calibration. Usually this is not desired, so the default is 1.0.
+        max_bl_cut: maximum baseline separation [meters] to keep in data when calibrating. None (default) means no maximum.
+        edge_cut: integer number of channels to exclude at each band edge in delay and global phase solvers
+        tol: baseline match tolerance in units of baseline vectors (e.g. meters)
+        phs_max_iter: integer maximum number of iterations of phase_slope_cal or TT_phs_cal allowed
+        phs_conv_crit: convergence criterion for updates to iterative phase calibration that compares them to all 1.0s.
+        clobber: if True, overwrites existing abscal calfits file at the output path
+        add_to_history: string to add to history of output abscal file
+
+    Returns:
+        hc: HERACal object which was written to disk. Matches the input redcal_file with an updated history and:
+            - gains: abscal gains for times that could be calibrated, redcal gains otherwise (but flagged)
+            - flags: redcal flags, with additional flagging if the data or model are flagged (see flag_utils.synthesize_ant_flags)
+            - quals: abscal chi^2 per antenna based on calibrated data minus model (Normalized by noise/nObs, but not with proper DoF)
+            - total_qual: abscal chi^2 based on calibrated data minus model (Normalized by noise/nObs, but not with proper DoF)
+    '''
+    # Raise error if output calfile already exists and clobber is False
+    if os.path.exists(redcal_file.replace(*output_replace)) and not clobber:
+        raise IOError("{} exists, not overwriting.".format(redcal_file.replace(*output_replace)))
+
```

Comment:
partial IO doesn't work for `miriad` files across time, so there should be a check on `nInt == None` if `filetype` == `miriad`